### PR TITLE
added new button component

### LINKED ISSUE #982: [NG] Improve Button component with new properties
**Current Behavior**
the current reusecore/Button requires wrapping an `<a>` around it, does not include a selectable color theme, require `alt` and `title` properties for the `<button>` or `<a>` tags.

**Desired Behavior**
Consider swapping for this component - https://www.npmjs.com/package/@hashicorp/react-button, which incorporates a number of useful properties; e.g.:

```
                  <Button
                    key={link.text}
                    title={link.text}
                    linkType={link.type}
                    url={link.url}
                    theme={{ variant, brand }}
                  />
```
---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by codetheorem**
Signed-off-by: Hrishikesh Agarwal <hrishikeshagarwalv@gmail.com>

**Description**
Added new button component according to this https://www.npmjs.com/package/@hashicorp/react-button.

This PR fixes #982 

**Notes for Reviewers**
I have added that button on the community page you can see that on /community route.

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: src/reusecore/NewButton/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    const initalButton = <BtnStyle className={addClasses.join(" ")}{...props} onClick={props.onClick}>{children}<img src={props.icon} />{props.title}</BtnStyle>;
+
+
+    return (
+        <React.Fragment>
+            { props.url && props.external &&       
+            <Link href={props.url} target="_blank" rel="noreferrer">{initalButton}</Link>}
+            {props.url && !props.external && 
+             <Link href={props.url}>{initalButton}</Link>}
```
Comment (Tanuj22) [inline src/reusecore/NewButton/index.js:27]: This will be `<Link to={props.url}>`

File: src/reusecore/NewButton/index.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        addClasses.push(className);
+    }
+
+    const initalButton = <BtnStyle className={addClasses.join(" ")}{...props} onClick={props.onClick}>{children}<img src={props.icon} />{props.title}</BtnStyle>;
+
+
+    return (
+        <React.Fragment>
+            { props.url && props.external &&       
+            <Link href={props.url} target="_blank" rel="noreferrer">{initalButton}</Link>}
```
Comment (Tanuj22) [inline src/reusecore/NewButton/index.js:25]: This will be `a` tag

**Review by Tanuj22 — COMMENTED** [review]
Are these all the functionalities we need @leecalcote? If Yes, we can replace all the existing buttons rather than creating a new component called `NewButton`

**Comment by leecalcote** [comment]
> Are these all the functionalities we need @leecalcote? If Yes, we can replace all the existing buttons rather than creating a new component called `NewButton`

@Tanuj22, yes, we'll want to enhance the existing component so that all buttons have the option of taking advantage of these parameters/functions.

**Comment by codetheorem** [comment]
@Tanuj22 I have created this component in a separate file so that after it is tested and finalized we can start replacing the current component with this one.

**Comment by Tanuj22** [comment]
> @Tanuj22 I have created this component in a separate file so that after it is tested and finalized we can start replacing the current component with this one.

LGTM, can you take it up in this PR for a separate PR?

**Review by leecalcote — APPROVED** [review]

**Pull request merged**